repositories/ingestion_repository.go

```diff
@@ -13,37 +13,38 @@ import (
 )
 
 type IngestionRepository interface {
-	IngestObjects(ctx context.Context, transaction Transaction_deprec, payloads []models.PayloadReader, table models.Table, logger *slog.Logger) (err error)
+	IngestObjects(ctx context.Context, exec Executor, payloads []models.PayloadReader, table models.Table, logger *slog.Logger) (err error)
 }
 
-type IngestionRepositoryImpl struct {
-}
+type IngestionRepositoryImpl struct{}
 
-func (repo *IngestionRepositoryImpl) IngestObjects(ctx context.Context, transaction Transaction_deprec, payloads []models.PayloadReader, table models.Table, logger *slog.Logger) (err error) {
-	tx := adaptClientDatabaseTransaction(transaction)
+func (repo *IngestionRepositoryImpl) IngestObjects(ctx context.Context, exec Executor, payloads []models.PayloadReader, table models.Table, logger *slog.Logger) (err error) {
+	if err := validateClientDbExecutor(exec); err != nil {
+		return err
+	}
 
 	mostRecentObjectIds, mostRecentPayloads := repo.mostRecentPayloadsByObjectId(payloads)
 
-	previouslyIngestedObjects, err := repo.loadPreviouslyIngestedObjects(ctx, tx, mostRecentObjectIds, table.Name)
+	previouslyIngestedObjects, err := repo.loadPreviouslyIngestedObjects(ctx, exec, mostRecentObjectIds, table.Name)
 	if err != nil {
 		return err
 	}
 
 	payloadsToInsert, obsoleteIngestedObjectIds := repo.comparePayloadsToIngestedObjects(mostRecentPayloads, previouslyIngestedObjects)
 
 	if len(obsoleteIngestedObjectIds) > 0 {
-		if err := repo.batchUpdateValidUntilOnObsoleteObjects(ctx, tx, table.Name, obsoleteIngestedObjectIds); err != nil {
+		if err := repo.batchUpdateValidUntilOnObsoleteObjects(ctx, exec, table.Name, obsoleteIngestedObjectIds); err != nil {
 			return err
 		}
 	}
 
 	if len(payloadsToInsert) > 0 {
-		if err := repo.batchInsertPayloadsAndEnumValues(ctx, tx, payloadsToInsert, table); err != nil {
+		if err := repo.batchInsertPayloadsAndEnumValues(ctx, exec, payloadsToInsert, table); err != nil {
 			return err
 		}
 	}
 
-	logger.Info("Inserted objects in db", slog.String("type", tableNameWithSchema(tx, table.Name)), slog.Int("nb_objects", len(payloadsToInsert)))
+	logger.Info("Inserted objects in db", slog.String("type", tableNameWithSchema(exec, table.Name)), slog.Int("nb_objects", len(payloadsToInsert)))
```

Comment:
Question unrelated with the scope of this PR: IIUC, we plan to use the logger from the `ctx` in the long term, and remove all injection like this one here ?

Reply:
yea absolutely, it's supposed to the case already. I'll clean that up, rather in another PR